rc/tools/make.kak

```diff
@@ -69,7 +69,7 @@ define-command -hidden make-jump %{
 define-command make-next-error -docstring 'Jump to the next make error' %{
     evaluate-commands -try-client %opt{jumpclient} %{
         buffer '*make*'
-        execute-keys "%opt{make_current_error_line}ggl" "/^(?:\w:)?[^:\n]+:\d+:(?:\d+:)?%opt{make_error_pattern}<ret>"
+        execute-keys "%opt{make_current_error_line}ggl" "/%opt{make_error_pattern}<ret>"
```

Comment:
`"/%opt{make_error_pattern}<ret>"` is prone to injection of special characters (e.g. if the pattern contains `<ret>`). To avoid it, you can split it in two lines
```
set-register / %opt{make_error_pattern}
execute-keys "%opt{make_current_error_line}ggl" "/<ret>"
```
As well as adding `-save-regs '/'` to the enclosing `evaluate-commands` call.

It's not as much of a problem for `make_current_error_line` since that's by definition an integer, which cannot contain any special character.

Reply:
I am not sure if we shall separate the fix into another PR, what do you think?

Reply:
We might as well do it as part of this one

Reply:
@occivink @mawww , it's done, please take a another look.

Reply:
this has some of the same problems as https://github.com/mawww/kakoune/pull/4972 (breaks compat, highlighting)
Also you didn't update `make-jump`.
If we want compatibility we could add a new option (`make_location_pattern`) and deprecate the old one.

Reply:
@krobelus , use a new option, please take another look, thanks.